### DISCUSSION TITLE
How to use retry to rerun the spec file after certain time or at the end of all the tests completed?

### DISCUSSION THREAD

**Discussion by Pverma96** [cypress-io/cypress#24949] [Questions and Help]
Currently when using the "retries" option the failed test cases are retried immediately after the failure. Is there a way by which the failed tests can run after certain amount of time or at the end once remaining tests execution is completed?

**Comment by portyu9** [comment]
Hi @Pverma96 

No. Cypress's built-in `retries` option does not have a setting to delay the retry or postpone it until all remaining tests/specs have finished.

A retry is another attempt of the same test, so Cypress runs it before continuing with the rest of the suite.

If you specifically need a "second pass" after everything else has completed, that needs to be implemented outside the normal retry mechanism — for example in your CI script or through Cypress's Node Module API by collecting failed specs and running those specs again afterward.

If the reason for wanting a delay is that the application/backend needs time to reach a particular state, I would avoid using a delayed test retry for that. Instead, wait for the specific condition inside the test (request completion, API state, DOM state, etc.), so the test does not depend on an arbitrary amount of time.

Please mark this answer as accepted if it helped, thank you.

**Comment by portyu9** [comment]
Hi @Pverma96 

Cypress supports uploading **files**, but it does not have a first-class command for selecting an entire directory from the browser's native "choose folder" UI.

For a normal multiple-file input you can pass an array to `selectFile()`:

```js
cy.get('input[type="file"][multiple]').selectFile([
  'cypress/fixtures/folder/file1.txt',
  'cypress/fixtures/folder/subfolder/file2.txt',
])
```

That attaches the files, but it is not equivalent to a real directory upload when the application depends on folder metadata such as `webkitRelativePath`. `cy.selectFile()` does not currently provide a documented way to set that directory-relative metadata.

For an application whose feature specifically uploads a folder hierarchy, the better options are usually:

- exercise the application's upload API directly with the expected paths/metadata, or
- upload an archive such as a `.zip` if that is what the application supports.

Downloads work similarly: browsers download **files**, not filesystem folders. Cypress stores downloaded files in `cypress/downloads` by default. If your application offers "download folder", the server would normally package it as something like a ZIP file, which you can then verify/extract in Node code if necessary.

Please mark this answer as accepted if it helped, thank you.
